File: web/src/branding.ts
/**
 * Single source of every user-visible product name on the web side.
 * Mirror of `Branding.java`. See `docs/adr/0001-project-name-and-trademark-risk.md`.
 *
 * A rename is: this file, `Branding.java`, the Maven `artifactId`, the image
 * coordinates, and `index.html` <title>. Nothing else.
 */
export const branding = {
  productName: 'Artemis Studio',
  productShortName: 'Studio',
  tagline:
    'Cluster-wide management and observability for Apache ActiveMQ Artemis',
  trademarkNotice:
    'Apache ActiveMQ and Apache ActiveMQ Artemis are trademarks of the Apache ' +
    'Software Foundation. Artemis Studio is an independent project and is not ' +
    'produced by, endorsed by, or affiliated with the Apache Software Foundation.',
  projectUrl: 'https://github.com/sudoitir/artemis-studio',
} as const;

File: src/main/java/io/github/sudoitir/artemisstudio/Branding.java
package io.github.sudoitir.artemisstudio;

/**
 * Single source of every user-visible product name and mark.
 *
 * <p><strong>Rename insurance.</strong> "Artemis Studio" applies an Apache Software
 * Foundation trademark (the bare project name {@code Artemis}) to this product.
 * ASF trademark policy discourages that, and precedent exists (Kafka Tool → Offset
 * Explorer). The decision to keep the name was made deliberately; see
 * {@code docs/adr/0001-project-name-and-trademark-risk.md}.
 *
 * <p>If the project is ever renamed, changing the constants here and their two
 * counterparts in {@code web/src/branding.ts}, plus the Maven {@code artifactId}
 * and image coordinates, is the entire job. Nothing else in the codebase should
 * hard-code the name.
 */
public final class Branding {

    private Branding() {}

    /** Full product name, as shown in titles and the About dialog. */
    public static final String PRODUCT_NAME = "Artemis Studio";

    /** Short name for compact UI (tab titles, breadcrumbs). */
    public static final String PRODUCT_SHORT_NAME = "Studio";

    public static final String TAGLINE = "Cluster-wide management and observability for Apache ActiveMQ Artemis";

    /**
     * Mandatory disclaimer. Rendered in the About dialog and the docs footer.
     */
    public static final String TRADEMARK_NOTICE = "Apache ActiveMQ and Apache ActiveMQ Artemis are trademarks of the "
            + "Apache Software Foundation. " + PRODUCT_NAME + " is an independent "
            + "project and is not produced by, endorsed by, or affiliated with the "
            + "Apache Software Foundation.";

    public static final String PROJECT_URL = "https://github.com/sudoitir/artemis-studio";
}
